lib/datadog/tracing/contrib/active_support/notifications/subscription.rb

```diff
@@ -109,18 +121,20 @@ def subscribers
             class Handler
               attr_reader :block
 
-              def initialize(&block)
+              def initialize(block)
                 @block = block
               end
 
               def run(span, name, id, payload)
-                run!(span, name, id, payload)
+                run!(span, name, id, payload) if @block
```

Comment:
I think it is more explicit to remove private method `#run!` and change.

```suggestion
                @block.call((span, name, id, payload) if @block
```

Reply:
Done!